-1

**Failed Tests**:   Build 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-a0938c/38653/summary.html
**COMMIT**: f1e8229ab78188b41be9f5f463a7e46c39c30ab8
**CMSSW**: CMSSW_14_1_X_2024-04-05-2300/el9_amd64_gcc13
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/44646/38653/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation error when building: 

<pre>/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02831/el9_amd64_gcc13/cms/cmssw/CMSSW_14_1_X_2024-04-05-2300/src/DataFormats/L1Trigger/interface/Muon.h:118:25: note:   by 'virtual bool l1t::Muon::operator!=(const l1t::Muon&) const'
  118 |     virtual inline bool operator!=(const l1t::Muon& rhs) const { return !(operator==(rhs)); };
      |                         ^~~~~~~~
>> Building edm plugin tmp/el9_amd64_gcc13/src/L1Trigger/L1TMuonBarrel/plugins/L1TriggerL1TMuonBarrelPlugins/libL1TriggerL1TMuonBarrelPlugins.so
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02831/el9_amd64_gcc13/external/gcc/13.2.0-1b0a3367d04f48f01ad3ccf40e55475c/bin/../lib/gcc/x86_64-redhat-linux-gnu/13.2.0/../../../../x86_64-redhat-linux-gnu/bin/ld.bfd: cannot find -lL1TriggerL1TMuonBarrel: No such file or directory
collect2: error: ld returned 1 exit status
gmake: *** [tmp/el9_amd64_gcc13/src/L1Trigger/L1TMuonBarrel/plugins/L1TriggerL1TMuonBarrelPlugins/libL1TriggerL1TMuonBarrelPlugins.so] Error 1
Leaving library rule at src/L1Trigger/L1TMuonBarrel/plugins
Entering library rule at L1Trigger/L1TMuonBarrel
>> Compiling  src/L1Trigger/L1TMuonBarrel/src/L1MuBMAssignmentUnit.cc
>> Compiling  src/L1Trigger/L1TMuonBarrel/src/L1MuBMDataBuffer.cc
</pre>

--- a/DQMOffline/Alignment/BuildFile.xml
+++ b/DQMOffline/Alignment/BuildFile.xml
@@ -15,3 +15,4 @@
 <use name="TrackingTools/TransientTrack"/>
 <use name="roofit"/>
 <flags EDM_PLUGIN="1"/>
+<flags CXXFLAGS="-Wno-error=array-bounds" />

--- a/L1Trigger/L1TMuonBarrel/BuildFile.xml
+++ b/L1Trigger/L1TMuonBarrel/BuildFile.xml
@@ -8,3 +8,4 @@
 <export>
   <lib name="1"/>
 </export>
+<flags CXXFLAGS="-Wno-error=array-bounds" />

--- a/RecoLocalCalo/HcalRecAlgos/BuildFile.xml
+++ b/RecoLocalCalo/HcalRecAlgos/BuildFile.xml
@@ -20,3 +20,4 @@
 <export>
   <lib name="1"/>
 </export>
+<flags CXXFLAGS="-Wno-error=array-bounds" />

